storage/file_adapter.go

```diff
@@ -12,18 +12,18 @@ import (
 
 var logRoot = "/data/logs"
 
-type adapter struct {
+type fileAdapter struct {
 	files map[string]*os.File
 	mutex sync.Mutex
 }
 
-// NewStorageAdapter returns a pointer to a new instance of a file-based storage.Adapter.
-func NewStorageAdapter() (*adapter, error) {
-	return &adapter{files: make(map[string]*os.File)}, nil
+// NewFileAdapter returns an Adapter that uses a file.
+func NewFileAdapter() (Adapter, error) {
```

Comment:
I'd like to offer custom constructors like this one alongside the factory method. I don't like factories but I figured I'd leave it around to avoid a bigger change in this patch